client/server/pages/index.js

```diff
@@ -227,8 +231,10 @@ function getDefaultContext( request, response, entrypoint = 'entry-main' ) {
 }
 
 const setupDefaultContext = ( entrypoint ) => ( req, res, next ) => {
-	req.context = getDefaultContext( req, res, entrypoint );
-	next();
+	getDefaultContext( req, res, entrypoint ).then( ( context ) => {
+		req.context = context;
+		next();
+	} );
```

Comment:
It's much more elegant to turn this function into an async function (`(entrypoint) => async (req, res, next) => ...`, and then the only change in the body is adding `await`:
```js
req.context = await getDefaultContext( ... );
```